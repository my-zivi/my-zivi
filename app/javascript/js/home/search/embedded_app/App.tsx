import algoliasearch from 'algoliasearch';
import React from 'preact/compat';
import { createBrowserLocalStorageCache } from '@algolia/cache-browser-local-storage';
import { createFallbackableCache } from '@algolia/cache-common';
import { createInMemoryCache } from '@algolia/cache-in-memory';
import aa from 'search-insights';
import Turbo from '@hotwired/turbo-rails';
import SearchPage from '~/js/home/search/embedded_app/components/SearchPage';

declare global {
  interface Window {
    Turbo: typeof Turbo;
  }
}

aa('init', {
  appId: MyZivi.algolia.applicationId,
  apiKey: MyZivi.algolia.apiKey,
  useCookie: true,
});

const searchClient = algoliasearch(
  MyZivi.algolia.applicationId,
  MyZivi.algolia.apiKey,
  {
    responsesCache: createFallbackableCache({
      caches: [
        createBrowserLocalStorageCache({ key: 'my-zivi-search', localStorage: window.sessionStorage }),
        createInMemoryCache(),
      ],
    }),
    requestsCache: createInMemoryCache({ serializable: false }),
  },
);

const App: React.FunctionComponent = () => (
  <SearchPage searchClient={searchClient} />
);

export default App;
