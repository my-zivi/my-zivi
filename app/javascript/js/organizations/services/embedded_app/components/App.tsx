import React from 'preact/compat';
import ApiProvider from '~/js/shared/ApiProvider';
import ServicesOverviewCard from '~/js/organizations/services/embedded_app/components/ServicesOverviewCard';

const App: React.FunctionComponent = () => (
  <ApiProvider>
    <ServicesOverviewCard />
  </ApiProvider>
);

export default App;
