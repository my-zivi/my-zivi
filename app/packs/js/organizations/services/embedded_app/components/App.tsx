import ApiProvider from 'js/shared/ApiProvider';
import ServicesOverviewCard from 'js/organizations/services/embedded_app/components/ServicesOverviewCard';
import React from 'preact/compat';

const App: React.FunctionComponent = () => (
  <ApiProvider>
    <ServicesOverviewCard />
  </ApiProvider>
);

export default App;
