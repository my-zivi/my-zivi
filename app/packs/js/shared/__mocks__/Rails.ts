/* eslint-disable @typescript-eslint/no-explicit-any */

const Rails = jest.createMockFromModule('@rails/ujs') as {
  ajax,
  mockError: (error: Error) => void,
  mockResponse: (response: any, forUrl: string) => void,
};

let returnedError = null;
const returnedResponses = {};

Rails.mockError = (error) => {
  returnedError = error;
};

Rails.mockResponse = (response, forUrl) => {
  returnedResponses[forUrl] = response;
};

interface AjaxOptions {
  success: (re: any) => void;
  error: (err: any) => void;
  url: string;
}

Rails.ajax = ({ success, error, url }: AjaxOptions): void => {
  if (returnedError) {
    error(returnedError);
  }
  success(url in returnedResponses ? returnedResponses[url] : null);
};

export default Rails;
