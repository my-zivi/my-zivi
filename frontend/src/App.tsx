import 'bootstrap/dist/css/bootstrap.min.css';
import 'izitoast/dist/css/iziToast.css';
import * as React from 'react';
import 'react-widgets/dist/css/react-widgets.css';

import { Route, Switch } from 'react-router-dom';
import { IziviLayout } from './layout/IziviLayout';

import { Icons } from './utilities/Icon';
import { ProtectedRoute } from './utilities/ProtectedRoute';
import { ChangeForgottenPassword } from './views/ChangeForgottenPassword';
import { ChangePassword } from './views/ChangePassword';
import { ExpenseSheetOverview } from './views/expense_sheets/ExpenseSheetOverview';
import { ExpenseSheetUpdate } from './views/expense_sheets/ExpenseSheetUpdate';
import { ForgotPassword } from './views/ForgotPassword';
import { HolidayOverview } from './views/holidays/HolidayOverview';
import { Home } from './views/Home';
import { Login } from './views/Login';
import { NotFound } from './views/NotFound';
import { PaymentDetail } from './views/payments/detail/PaymentDetail';
import { PaymentOverview } from './views/payments/PaymentOverview';
import { PhoneListView } from './views/PhoneList';
import { Register } from './views/register/Register';
import { ServiceOverview } from './views/service_overview/ServiceOverview';
import { ServiceSpecificationsOverview } from './views/service_specification/ServiceSpecificationsOverview';
import { UserFeedbackOverview } from './views/user_feedback_overview/UserFeedbackOverview';
import { ProfileOverview } from './views/users/ProfileOverview';
import { ServiceFeedback } from './views/users/service_feedback/ServiceFeedback';
import { UserOverview } from './views/users/UserOverview';
import { UserUpdate } from './views/users/UserUpdate';

Icons();

class App extends React.Component {
  render() {
    return (
      <IziviLayout>
        <Switch>
          <Route component={Home} exact path={'/'} />
          <Route component={Login} exact path={'/login'} />
          <Route component={Register} exact path={'/register/:page'} />
          <Route component={ForgotPassword} exact path={'/users/password/reset'} />
          <Route component={ChangeForgottenPassword} exact path={'/users/password/edit/:reset_password_token'} />
          <Route component={HolidayOverview} exact path={'/holidays'} />
          <Route component={PhoneListView} exact path={'/phones'} />
          <Route component={ProfileOverview} exact path={'/profile'} />
          <ProtectedRoute component={ChangePassword} exact path={'/changePassword'} />
          <ProtectedRoute requiresAdmin component={PaymentOverview} exact path={'/payments'} />
          <ProtectedRoute requiresAdmin component={ServiceOverview} exact path={'/services'} />
          <ProtectedRoute requiresAdmin component={PaymentDetail} exact path={'/payments/:timestamp'} />
          <ProtectedRoute requiresAdmin component={ExpenseSheetOverview} exact path={'/expense_sheets'} />
          <ProtectedRoute requiresAdmin component={ExpenseSheetUpdate} exact path={'/expense_sheets/:id'} />
          <ProtectedRoute requiresAdmin component={UserFeedbackOverview} exact path={'/user_feedbacks'} />
          <ProtectedRoute requiresAdmin component={UserOverview} exact path={'/users'} />
          <ProtectedRoute requiresAdmin component={UserUpdate} exact path={'/users/:id'} />
          <ProtectedRoute requiresAdmin component={ServiceSpecificationsOverview} exact path={'/service_specifications'} />
          <Route component={NotFound} />
        </Switch>
      </IziviLayout>
    );
  }
}

export default App;
