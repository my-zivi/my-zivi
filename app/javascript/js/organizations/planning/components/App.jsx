import React, { h, Component, Fragment } from 'preact';
import Rails from '@rails/ujs';
import { PlanningTable } from './PlanningTable';
import ServicesPlan from '../models/ServicesPlan';

export default class App extends Component {
  constructor() {
    super();

    this.state = {
      loading: true,
      services: [],
    };
  }

  async componentDidMount() {
    await this.reloadPlanningView();
  }

  async reloadPlanningView() {
    const services = await this.fetchData();

    this.setState({
      loading: false,
      services,
    });
  }

  async fetchData() {
    return new Promise((success, error) => {
      Rails.ajax({
        url: '/organizations/planning.json',
        type: 'GET',
        success,
        error,
      });
    });
  }

  servicesPlan() {
    return new ServicesPlan(this.state.services);
  }

  render() {
    if (this.state.loading) {
      return <p>loading...</p>;
    } else {
      const plan = this.servicesPlan();

      return(
        <>
          <h1>{plan.planBeginning} - {plan.planEnding}</h1>
          <PlanningTable servicesPlan={plan} />
        </>
      );
    }
  }
}
