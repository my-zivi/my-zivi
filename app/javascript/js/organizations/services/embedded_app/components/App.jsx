import React, { h, Component, Fragment } from 'preact';
import Rails from '@rails/ujs';
import { PlanningTable } from './PlanningTable';
import ServicesPlan from '../models/ServicesPlan';
import { DATE_FORMATS } from '../../../../constants';

export default class App extends Component {
  constructor() {
    super();

    this.state = {
      loading: true,
      services: [],
    };
  }

  async componentDidMount() {
    await this.reloadPlanningView();
  }

  async reloadPlanningView() {
    const services = await this.fetchData();

    this.setState({
      loading: false,
      services,
    });
  }

  async fetchData() {
    return new Promise((success, error) => {
      Rails.ajax({
        url: '/organizations/services.json',
        type: 'GET',
        success,
        error,
      });
    });
  }

  servicesPlan() {
    return new ServicesPlan(this.state.services);
  }

  render() {
    if (this.state.loading) {
      return <p>loading...</p>;
    } else {
      const plan = this.servicesPlan();

      return(
        <>
          <h1>{plan.planBeginning.format(DATE_FORMATS.short)} - {plan.planEnding.format(DATE_FORMATS.short)}</h1>
          <PlanningTable servicesPlan={plan} />
        </>
      );
    }
  }
}
