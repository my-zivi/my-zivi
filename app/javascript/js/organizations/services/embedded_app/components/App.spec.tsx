import App from 'js/organizations/services/embedded_app/components/App';
import { shallow } from 'enzyme';
import React from 'preact/compat';
import Rails from 'js/shared/__mocks__/Rails';
import Factories from 'js/tests/factories/Factories';
import ServiceFactory from 'js/tests/factories/ServiceFactory';

describe('App', () => {
  it('runs', () => {
    Rails.mockResponse(Factories.buildList(ServiceFactory, 2), MyZivi.paths.servicesOverview);
    const wrapper = shallow(<App />);
    wrapper.update();

    expect(wrapper).toMatchSnapshot();
  });
});
