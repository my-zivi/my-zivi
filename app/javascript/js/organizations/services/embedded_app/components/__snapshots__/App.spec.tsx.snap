// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App runs 1`] = `
<div
  className="card mb-3 mb-lg-0"
>
  <CardHeader
    overviewSubtitle="Laden..."
  />
  <div
    className="card-body"
  >
    <Spinner
      style="info"
    />
  </div>
</div>
`;
